src/registry2.0/portimage.tcl

```diff
@@ -348,9 +348,18 @@ proc extract_archive_to_tmpdir {location} {
             t(ar|bz|lz|xz|gz) {
                 set tar "tar"
                 if {[catch {set tar [macports::findBinary $tar ${macports::autoconf::tar_path}]} errmsg] == 0} {
```

Comment:
It's used to set the error string if finding tar fails. Leaving it for now.

Reply:
It makes no sense that you would error out when `tar` was not found, but the code would actually never have used it.

I meant it should be like this:
```
if {![catch {macports::binaryInPath bsdtar}] && ... } {
    ... # using bsdtar --hfsCompression
} else {
    if {[catch {set tar [macports::findBinary tar ...] ... } {
        ... # using tar
    } else {
        ... # tar not found error
    }
}
```

Reply:
The tar string set in 355 is used both in 356 (where it may get set again), and again in all the way down in 411 if finding tar fails. Yes, this could be rewritten with the throw much closer to the checks, and the "success case" outside the if - else - throw structure, but I was trying to minimize the lines of code (and whitespace from a removed indent) change... your thoughts? I can go either way.